src/main/java/seedu/address/logic/parser/ProfileCommandParser.java

```diff
@@ -0,0 +1,50 @@
+package seedu.address.logic.parser;
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ROLE;
+
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.UserProfile;
+import seedu.address.logic.commands.ProfileCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new ProfileCommand object
+ */
+public class ProfileCommandParser implements Parser<ProfileCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the {@code ProfileCommand}
+     * and returns a {@code ProfileCommand} object for execution.
+     * @throws ParseException if the user input does not conform to the expected format
+     */
+    public ProfileCommand parse(String args) throws ParseException {
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_ROLE);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_ROLE)
+                || !argMultimap.getPreamble().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ProfileCommand.MESSAGE_USAGE));
+        }
+
+        try {
+            String name = argMultimap.getValue(PREFIX_NAME).get();
+            UserProfile.Role role = UserProfile.Role.valueOf(argMultimap.getValue(PREFIX_ROLE).get().toUpperCase());
```

Comment:
This line is also slightly long and hard to understand, maybe refactor this?

Reply:
Will do. Thanks!